### DISCUSSION TITLE
Print sympy expression from Aesara

### DISCUSSION THREAD

**Discussion by gkaissis** [aesara-devs/aesara#493] [Ideas]
Congratulations on the great project!
Is it possible to obtain a sympy expression for a given computational graph? Obviously not all aesara primitives map to sympy, but it would certainly be interesting to be able to "parse" the aesara expression tree to a regular "formula" to be able to reason about certain processes analytically, even for a restricted subset of aesara. In a way, I am looking for an "inverse code generator"

**Comment by twiecki** [comment]
There have been a few attempts of this over the years. It does work OKish, but the main problem is with higher-dimensional operations like slicing.

**Comment by gkaissis** [comment]
Are there any options for actually doing it (even incomplete ones)? I have not been able to find anything in the API. Thanks a lot!

**Comment by twiecki** [comment]
Google for Theano and Sympy.

**Comment by twiecki** [comment]
https://docs.sympy.org/latest/modules/numeric-computation.html#aesara

**Comment by gkaissis** [comment]
I'm with you on the SymPy -> Aesara (Theano) side, but I don't see the reverse (Aesara -> SymPy) on the SymPy docs (just found some blogs from 2013). I'm happy to look further myself or write a parser for the stuff I need, but if you had any specific pointers, I'd be grateful!
Many thanks again!

**Comment by twiecki** [comment]
I see, yeah I don't think there's recent stuff on that.
